unic/ucd/name/src/name.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+use std::fmt;
+
+use unic_utils::CharDataTable;
+
+
+#[derive(Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
```

Comment:
What does `PartialOrd` do here? Does it really implement an ordering that derefs all the strings and comparing them using `str`'s ordering?

I'm getting more conscious about deriving `*Ord` traits now. They are not in the contracts, and maybe we should actually not derive them unless we know what we want and we add unit tests for them?

Also, this now reminds me that this module needs at least a minimal unit test to check a couple of basic entries and default values (`None`).

I'll work on the contract part after landing, so don't worry about that part.